Lib/imaplib.py

```diff
@@ -1279,6 +1385,236 @@ def print_log(self):
                 n -= 1
 
 
+class _Idler:
+    # Iterable context manager: start IDLE & produce untagged responses
+    #
+    # This iterator produces (type, datum) tuples.  They slightly differ
+    # from the tuples returned by IMAP4.response():  The second item in the
+    # tuple is a single datum, rather than a list of them, because only one
+    # untagged response is produced at a time.
```

Comment:
Should be in a string:
```suggestion
    """Iterable context manager: start IDLE & produce untagged responses
    
    This iterator produces (type, datum) tuples.  They slightly differ
    from the tuples returned by IMAP4.response():  The second item in the
    tuple is a single datum, rather than a list of them, because only one
    untagged response is produced at a time."""
```

Reply:
Hmm. This is an interesting case.

The `_Idler` class is deliberately an internal type rather than a public one, because we don't want users to subclass it, instantiate it, depend on its structure, or otherwise directly reference it in any way. Documenting it as a type would be [promoting it to public API](https://peps.python.org/pep-0008/#public-and-internal-interfaces) status, thereby making future changes difficult due to the need for backward compatibility. (Note that documentation on how to use it *does* exist in the docs; just not as an object to be instantiated by users.)

On the other hand, it contains a public method with a doc string. I can therefore see an argument for the containing class to have a doc string as well, and perhaps a stable name to make finding those doc strings easy with the `help()` builtin.

It's a bit of a trade-off. With both those things in mind, do you still prefer the latter option?

Reply:
On further consideration, I could mitigate the problems with making it public by declaring it "provisional or internal" in the docs, as suggested in PEP 8. Unless you have a better suggestion, I think I'll do that.

Reply:
I think we do need to document it (see my other comment). It's going to have to go into typeshed anyway, we can't easily just pretend it doesn't exist in the future.

Reply:
Done in 94c02e88a9d3.